.travis.yml

```diff
@@ -18,26 +19,34 @@ matrix:
       os: linux
     - name: "OSX"
       os: osx
+    - name: CentOS 7
+      os: linux
+      env: IMAGE=centos7 INDOCKER="docker exec target"
+
+services:
+- docker
 
 branches:
   only:
     - master
     - /^v\d+\.\d+\.\d+/
 
 before_install:
+  - ./.travis_scripts/maybe-start-docker.sh
+  - ${INDOCKER} ./.travis_scripts/maybe-install-rust.sh
   - openssl aes-256-cbc -K $encrypted_487cf28eaba8_key -iv $encrypted_487cf28eaba8_iv
     -in tests/src/test/resources/service-keys.conf.enc -out tests/src/test/resources/service-keys.conf -d
 install:
-  - "~/bin/sbt --help || curl -Ls https://git.io/sbt > ~/bin/sbt && chmod 0755 ~/bin/sbt"
+  - ${INDOCKER} ./.travis_scripts/maybe-install-sbt.sh
```

Comment:
is `INDOCKER` a Travis provided thing? What does it do?

Reply:
It's defined in this file, in the build matrix. If we're running in a container, it expands to a `docker exec`; otherwise, it's empty.